### PR TITLE
cleanup unused --no-mul-mat-q,-nommq, -mmq, --mul-mat-q, mul_mat_q

### PR DESCRIPTION
**Pull request by phymbert**
Cleanup deprecated not used `params.mul_mat_q` and related docs. It can make confusion for user trying to optmize GPU inference performance: #3359 and #3412 .

Breaking upstream application if any using it.

### CODE REVIEW

**Comment by slaren** [comment]
This option is no longer supported, and should be removed completely instead.

**Comment by phymbert** [comment]
@slaren Thanks, I did not checked :/, I will clean it up it then.

File: examples/batched-bench/batched-bench.cpp [hunk 1]
```diff
@@ -41,7 +41,6 @@ int main(int argc, char ** argv) {
     int n_kv_max     = 2048;
     int is_pp_shared = 0;
     int n_gpu_layers = 0;
-    int mmq          = 0;
```
Comment (ggerganov) [inline examples/batched-bench/batched-bench.cpp:44]: Also remove from the `usage` prints above

Comment (phymbert) [reply examples/batched-bench/batched-bench.cpp:44]: Done. @ggerganov By the way what is the HF repo and file to run the `llama-bench`  please ?

```shell
./llama-bench -o sql | sqlite3 llama-bench.sqlite
main: error: failed to load model 'models/7B/ggml-model-q4_0.gguf'
```

**Review by slaren — APPROVED** [review]

File: examples/batched-bench/batched-bench.cpp (hunk 1 above)
Comment (slaren) [reply examples/batched-bench/batched-bench.cpp:44]: You can use any model with `llama-bench`, the default is supposed to be llama-7B Q4_0, but it is not important to use that model.

**Comment by phymbert** [comment]
@ggerganov Hi, as I got one approval, should I wait for yours ? thanks

**Review by ggerganov — APPROVED** [review]

**Pull request merged**